### PR TITLE
Merged PR: Tileset export and User Data

### PR DESCRIPTION
**Pull request by B-Reif**
This PR addresses some merge conflicts with the other two PRs I have opened. It also includes a fix for an edge case when tilemap cels overflow the image bounds.

### CODE REVIEW

File: src/file.rs [hunk 1]
```diff
@@ -204,11 +216,45 @@ impl AsepriteFile {
...
+        Ok(tileset.write_to_image(image_pixels.as_ref()))
+    }
+
+    /// Access the user data for the entire sprite, if any exists.
+    pub fn sprite_user_data(&self) -> Option<&UserData> {
+        self.sprite_user_data.as_ref()
+    }
+
+    /// Access the file's [Slice]s.
+    pub fn slices(&self) -> &Vec<Slice> {
```
Comment (alpine-alpaca) [inline src/file.rs:254]: That should return a `&[Slice]` not a borrow of a vector.

File: src/file.rs [hunk 2]
```diff
@@ -428,11 +439,7 @@ fn blend_mode_to_blend_fn(mode: BlendMode) -> BlendFn {
     }
 }
 
-fn tile_pixels<'a>(
-    pixels: &'a [pixel::Rgba],
-    tile_size: &TileSize,
-    tile_id: &TileId,
-) -> &'a [pixel::Rgba] {
+fn tile_slice<'a, T>(pixels: &'a [T], tile_size: &TileSize, tile_id: &TileId) -> &'a [T] {
```
Comment (alpine-alpaca) [inline src/file.rs:442]: A comment explaining the intent of this function would be nice. It's not that obvious from the name & type.

**Review by alpine-alpaca — CHANGES_REQUESTED** [review]
That's one big PR. Took me 1h to review.

It's looking good though. And looks to be close to feature complete.

**Comment by alpine-alpaca** [comment]
Once this is merged I'll make a proper 0.3 release.

Hopefully, that should help with your other package.

**Comment by alpine-alpaca** [comment]
Actually, hold on making any changes. I'm doing the cleanups myself in branch `pr-9`.

**Comment by B-Reif** [comment]
Thanks for all your help on these features, I really appreciate it!

**Pull request merged**